src/recipe/recipe.service.ts

```diff
@@ -24,8 +25,24 @@ export class RecipeService {
     }
 
     return recipe;
+  }
 
   fetchAllRecipes(): Promise<Recipe[]> {
     return this.prisma.recipe.findMany();
   }
+
+  async updateRecipe(
+    id: number,
+    payload: UpdateRecipeRequest,
+  ): Promise<Recipe> {
+    try {
+      const recipe = await this.prisma.recipe.update({
+        where: { id },
+        data: payload,
+      });
+      return recipe;
+    } catch {
+      throw new NotFoundException();
+    }
```

Comment:
yep, in this case awaiting is justified by the try catch block 👍🏻 good job 👍🏻 

Later when we introduce repository / dao we will probably return prisma call immediately and then in service await and try catch it